test/unittest/transport/can/test_can_msg_rx_session.cpp

```diff
@@ -0,0 +1,144 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#include <libcyphal/transport/can/transport.hpp>
+
+#include "media_mock.hpp"
+#include "../multiplexer_mock.hpp"
+#include "../../gtest_helpers.hpp"
+#include "../../memory_resource_mock.hpp"
+#include "../../tracking_memory_resource.hpp"
+
+#include <gmock/gmock.h>
+
+namespace
+{
+using namespace libcyphal;
+using namespace libcyphal::transport;
+using namespace libcyphal::transport::can;
+
+using testing::_;
+using testing::Eq;
+using testing::Return;
+using testing::IsNull;
+using testing::NotNull;
+using testing::Optional;
+using testing::StrictMock;
+using testing::VariantWith;
+
+using namespace std::chrono_literals;
+
+class TestCanMsgRxSession : public testing::Test
+{
+protected:
+    void TearDown() override
+    {
+        // TODO: Uncomment this when PMR deleter is fixed.
+        // EXPECT_EQ(mr_.total_allocated_bytes, mr_.total_deallocated_bytes);
```

Comment:
Due to `cetl::pmr::Factory::make_unique` limitation, namely incorrect `size_bytes` at `cetl::pmr::memory_resource::deallocate`.